instrumentation/opentelemetry-instrumentation-logging/README.rst

```diff
@@ -13,6 +13,140 @@ Installation
 
     pip install opentelemetry-instrumentation-logging
 
+Log record enrichment behavior of logging instrumentation
+
+"""
+The OpenTelemetry ``logging`` integration automatically injects tracing context into log statements.
+It's structured within an `if` statement that checks  [`set_logging_format`] is `True`, indicating that logging configuration should be applied.
+
+The integration registers a custom log record factory with the the standard library logging module that automatically inject
+tracing context into log record objects. Optionally, the integration can also call ``logging.basicConfig()`` to set a logging
+format with placeholders for span ID, trace ID and service name.
+
+1. **Determining the Log Format**: The code first attempts to determine the logging format by looking for a `logging_format` key in the [`kwargs`]( dictionary. If it's not found, it tries to fetch the value from an environment variable named [`OTEL_PYTHON_LOG_FORMAT`]. If neither is available, it defaults to a predefined constant [`DEFAULT_LOGGING_FORMAT`]. This approach provides flexibility, allowing the log format to be specified through function arguments, environment variables, or falling back to a default if neither is provided.
+
+2. **Determining the Log Level**: Similarly, the log level is determined by looking for a [`log_level`] key in [`kwargs`](. If not found, it attempts to fetch a value from an environment variable [`OTEL_PYTHON_LOG_LEVEL`] and maps this value through a [`LEVELS`] dictionary to get the corresponding log level. If this process doesn't yield a result, it defaults to [`logging.INFO`]. This mechanism allows for dynamic adjustment of the log level based on runtime configurations or environment settings.
+
+3. **Applying the Configuration**: Finally, with the determined [`log_format`] and [`log_level`](c, the code calls [`logging.basicConfig(format=log_format, level=log_level)`]. This function is a part of Python's standard logging library and is used to configure the basic settings of the logging system. It sets up the default format for log messages and the log level, which controls the severity of the messages that are processed.
+
+The following keys are injected into log record objects by the factory:
+
+- ``otelSpanID``
+- ``otelTraceID``
+- ``otelServiceName``
+- ``otelTraceSampled``
+
+The integration uses the following logging format by default:
+
+.. code-block::
+
+    {default_logging_format}
```

Comment:
No need to have a separate section for the default format. Simply paste it into the below section regarding `OTEL_PYTHON_LOG_FORMAT` env var